src/Javis.jl

```diff
@@ -647,15 +646,17 @@ function javis(
         filecounter += 1
     end
 
-    !creategif && return 
-    # generate a colorpalette first so ffmpeg does not have to guess it
-    ffmpeg_exe(`-loglevel panic -i $(tempdirectory)/%10d.png -vf "palettegen=stats_mode=diff" -y "$(tempdirectory)/palette.bmp"`)
-    # then apply the palette to get better results
-    ffmpeg_exe(`-loglevel panic -framerate $framerate -i $(tempdirectory)/%10d.png -i "$(tempdirectory)/palette.bmp" -lavfi "paletteuse=dither=sierra2_4a" -y $pathname`)
-    !deletetemp && return 
-    run(`rm -rf $(tempdirectory)`)
-    mkpath(tempdirectory)
-    nothing
+    isempty(pathname) && return
+    path, ext = splitext(pathname)
+    if ext == ".gif"
+        # generate a colorpalette first so ffmpeg does not have to guess it
+        ffmpeg_exe(`-loglevel panic -i $(tempdirectory)/%10d.png -vf "palettegen=stats_mode=diff" -y "$(tempdirectory)/palette.bmp"`)
+        # then apply the palette to get better results
+        ffmpeg_exe(`-loglevel panic -framerate $framerate -i $(tempdirectory)/%10d.png -i "$(tempdirectory)/palette.bmp" -lavfi "paletteuse=dither=sierra2_4a" -y $pathname`)
+    else
+        @error "Currently it's only supported to create a gif and not a $ext."
+    end
```

Comment:
English grammar mistake here. It should read:

> Currently, only gif creation is supported and not a $ext.

Reply:
Oh yeah and I should add a test case for that...